packages/server/src/api/routes/tests/search.spec.ts

```diff
@@ -2133,8 +2142,47 @@ describe.each([
             related1: [{ _id: relatedRows[0]._id }],
             related2: [{ _id: relatedRows[1]._id }],
           },
+          {
+            name: "test2",
+            related1: [{ _id: relatedRows[2]._id }],
+            related2: [{ _id: relatedRows[3]._id }],
+          },
         ])
       })
+
+      isSqs &&
```

Comment:
Are these tests not applicable to all datasources?

Reply:
Lucene can't do this and I've limited these test cases to internal because this doesn't have a lot of impact on external DBs, this is a problem which is specific to the internal implementation of relationships (and it's odd use of junction tables for efficiency)

Reply:
I'd argue that while this problem affects SQS today, there's no reason it won't affect external datasources later. I'd strongly lean toward enabling this test in as many places as we can, to ensure parity is being maintained and future regressions don't affect places we weren't expecting.